client/allocrunner/taskrunner/plugin_supervisor_hook.go

```diff
@@ -280,7 +299,9 @@ WAITFORREADY:
 	for {
 		select {
 		case <-startCtx.Done():
-			h.kill(ctx, fmt.Errorf("CSI plugin failed probe: %v", err))
+			h.restartTask(ctx, fmt.Errorf("CSI plugin failed probe: %v", err))
```

Comment:
I don't think it really matters here, but I always try to wrap errors where possible to retain the error type (mostly for testing purposes).

```suggestion
			h.restartTask(ctx, fmt.Errorf("CSI plugin failed probe: %w", err))
```

Reply:
Good catch... I've made a pass over the whole file. This would be a good semgrep lint if we could figure one out.